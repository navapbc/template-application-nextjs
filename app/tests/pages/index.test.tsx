import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import Index from "src/pages/index";

describe("Index", () => {
  // Demonstration of rendering translated text, and asserting the presence of a dynamic value.
  // You can delete this test for your own project.
  it("renders link to Next.js docs", () => {
    render(<Index />);

    const link = screen.getByRole("link", { name: /next\.js/i });

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "https://nextjs.org/docs");
  });

  it("passes accessibility scan", async () => {
    const { container } = render(<Index />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
