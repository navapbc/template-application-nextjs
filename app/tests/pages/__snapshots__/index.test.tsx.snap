// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render the heading 1`] = `
<h1>
  Welcome to your 
  <a
    href="https://github.com/navapbc/template-application-nextjs"
  >
    Next.js Template!
  </a>
</h1>
`;
